### Categorical / Identifier Variables

In [1]:
%run '../lib/libraries.ipynb'

dython              0.6.1
tinydb              4.2.0


You should consider upgrading via the '/Users/duane.hinkley/PycharmProjects/c772-capstone-project/venv/bin/python -m pip install --upgrade pip' command.
You should consider upgrading via the '/Users/duane.hinkley/PycharmProjects/c772-capstone-project/venv/bin/python -m pip install --upgrade pip' command.


In [2]:
filterDf  = load_df('filterDf')
descDf = load_df('descDf')
types = get_var_types()

#### Variable List

In [3]:
variable_types(descDf, 'Categorical Identifier')

category                          variable
0           Assessment (a)                     assessment_id
1           Assessment (a)            assessment_instance_id
2  Assignment Attempt (aa)    assessment_instance_attempt_id
3  Assignment Attempt (aa)     learner_assignment_attempt_id
4        Item Attempt (ia)       assessment_item_response_id
5        Item Attempt (ia)  learner_assigned_item_attempt_id
6              Learner (l)                        learner_id
7         Organization (o)                            org_id
8              Section (s)                        section_id

#### Counts

In [4]:
unique_nulls(filterDf, types['identifierVars'])

null  unique
assessment_id                         0     329
assessment_instance_attempt_id     3264    8483
assessment_instance_id                0     615
assessment_item_response_id       15710   64368
learner_assigned_item_attempt_id      0   79689
learner_assignment_attempt_id         0    8855
learner_id                            0    1126
org_id                                0       3
section_id                            0      46

Totals
- org_id: 3
  - ID per organization
- section_id: 46
  - ID per section/school
- learner_id: 1126
  - ID per learner/student
- assessment_id: 329
  - ID per each assessment/quiz.
- assessment_instance_id: 615
  - ID of each time a section uses the assessment
- assessment_instance_attempt_id: 8483 null: 3264
  - ID of each time an assessment is attempted by a student
- assessment_item_response_id: 64368 null: 15710
  - ID of each response/answer to an assessment question
- learner_assignment_attempt_id: 8855
  - ID of each learner attempt of an assigned assessment
- learner_assigned_item_attempt_id: 79689
  - ID of each learner attempt of a item/question on an assigned assessment

Nulls need further investigation

In [5]:
add_todo('Investigate 15710 null assessment_item_response_id')


'Todo: Investigate 15710 null assessment_item_response_id'